### PR TITLE
Feat: Create Okta OAuth Adapter

### PR DESCRIPTION
**Pull request by tanay1337**
## What does this PR do?

Adds Okta OAuth Adapter

## Test Plan

* Enable Okta in the Users Settings tab
* Manually try logging in with Okta

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/410

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by adityaoberai** [comment]
Hey @tanay1337 
Was just looking at the adapter. I think it might be a good idea to add another custom input for the Authorization Server Id as well and only let that remain as `default` if left empty. What do you think?

**Comment by tanay1337** [comment]
That makes sense because, in its current state, the adapter would be useless for custom Authorization Server IDs @adityaoberai.

**Comment by christyjacob4** [comment]
@tanay1337  This looks great! 👏  Awesome work.
Could you also attach a gif / screenshots of a successful flow ?

**Comment by tanay1337** [comment]
Sure @christyjacob4!

Screenshot 2022-04-26 at 3 56 29 PM

Screenshot 2022-04-26 at 3 57 29 PM

As suggested in the provider docs, I did a manual test that resulted in a successful redirect back to appwrite.

File: app/views/console/users/oauth/okta.phtml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<label for="oauth2<?php echo $this->escape(ucfirst($provider)); ?>ClientSecret">Client Secret <span class="tooltip" data-tooltip="Provided in the Application you created in Okta"><i class="icon-info-circled"></i></span></label>
+<input name="clientSecret" id="oauth2<?php echo $this->escape(ucfirst($provider)); ?>ClientSecret" type="password" autocomplete="off" placeholder="Client Secret" />
+<label for="oauth2<?php echo $this->escape(ucfirst($provider)); ?>Domain">Okta Domain<span class="tooltip" data-tooltip="Your Okta Domain (without 'https://')"><i class="icon-info-circled"></i></span></label>
+<input name="oktaDomain" id="oauth2<?php echo $this->escape(ucfirst($provider)); ?>Domain" type="text" autocomplete="off" placeholder="dev-1337.okta.com" />
+<label for="oauth2<?php echo $this->escape(ucfirst($provider)); ?>AuthorizationServerId">Authorization Server ID<span class="tooltip" data-tooltip="Authorization Server ID for custom authorization servers"><i class="icon-info-circled"></i></span></label>
+<input name="authorizationServerId" id="oauth2<?php echo $this->escape(ucfirst($provider)); ?>AuthorizationServerId" type="text" autocomplete="off" placeholder="default" />
```
Comment (christyjacob4) [inline app/views/console/users/oauth/okta.phtml:12]:
```suggestion
<input name="authorizationServerId" id="oauth2<?php echo $this->escape(ucfirst($provider)); ?>AuthorizationServerId" type="text" autocomplete="off" placeholder="Default" />
```

Comment (christyjacob4) [reply app/views/console/users/oauth/okta.phtml:12]: Alright got it 👍

**Pull request merged**